File: IARA.ISS.Web/IARA.Web/src/app/shared/notifications/notification.ts
import { BaseNotification } from './base-notification';
import { NotificationTypes } from './notification-types.enum';

export class Notification<T> extends BaseNotification {

    public constructor(obj?: Partial<Notification<T>>) {
        super(NotificationTypes.User, undefined);
        Object.assign(this, obj);
    }

    public get message(): T {
        return this._message as T;
    }

    public set message(value: T) {
        this._message = value;
    }
}

File: IARA.ISS.Web/IARA.Notifications/Models/Notification.cs
using IARA.Notifications.Enums;

namespace IARA.Notifications.Models
{
    public class Notification<T> : BaseNotification
    {
        public Notification(NotificationTypes type)
        {
            Type = type;
        }

        public new T Message
        {
            get
            {
                return (T)base.Message;
            }
            set
            {
                base.Message = value;
            }
        }
    }
}
